src/pycram/datastructures/world_entity.py

```diff
@@ -110,10 +120,275 @@ def remove_saved_states(self) -> None:
+
+    @is_translating.setter
+    def is_translating(self, is_translating: bool) -> None:
+        self._is_translating = is_translating
+
+    @property
+    def is_rotating(self) -> Optional[bool]:
+        return self._is_rotating
+
+    @is_rotating.setter
+    def is_rotating(self, is_rotating: bool) -> None:
+        self._is_rotating = is_rotating
+
+    @property
+    def position(self) -> Point:
+        """
+        :return: A Point object containing the position of the link relative to the world frame.
+        """
+        return self.pose.position
+
+    @property
+    def position_as_list(self) -> List[float]:
+        """
+        :return: A list containing the position of the link relative to the world frame.
+        """
+        return self.pose.position_as_list()
+
+    @property
+    def orientation(self) -> Quaternion:
+        """
+        :return: A Quaternion object containing the orientation of the link relative to the world frame.
+        """
+        return self.pose.orientation
+
+    @property
+    def orientation_as_list(self) -> List[float]:
+        """
+        :return: A list containing the orientation of the link relative to the world frame.
+        """
+        return self.pose.orientation_as_list()
+
+    @property
+    def pose_as_list(self) -> List[List[float]]:
+        """
+        :return: A list containing the position and orientation of the link relative to the world frame.
+        """
+        return self.pose.to_list()
+
+    @property
+    def transform(self) -> Transform:
+        """
+        The transform of this entity.
+
+        :return: The transform of this entity.
+        """
+        return self.pose.to_transform(self.tf_frame)
+
+    def update_transform(self, transform_time: Optional[Time] = None) -> None:
+        """
+        Update the transformation of this link at the given time.
+
+        :param transform_time: The time at which the transformation should be updated.
+        """
+        self.local_transformer.update_transforms([self.transform], transform_time)
+
+    @property
+    @abstractmethod
+    def pose(self) -> Pose:
+        """
+        :return: A Pose object containing the pose of the link relative to the world frame.
+        """
+        ...
+
+    @pose.setter
+    @abstractmethod
+    def pose(self, pose: Pose) -> None:
+        """
+        Set the pose of this body.
+
+        :param pose: The pose of this body.
+        """
+        ...
+
+    @property
+    @abstractmethod
+    def tf_frame(self) -> str:
+        """
+        The tf frame of this entity.
+
+        :return: The tf frame of this entity.
+        """
+        pass
+
+    @property
+    def contact_points(self) -> ContactPointsList:
+        """
+        :return: The contact points of this body with other physical bodies.
+        """
+        return self.world.get_body_contact_points(self)
+
+    def get_contact_points_with_body(self, body: 'PhysicalBody') -> ContactPointsList:
```

Comment:
import from __future__ import annotations and remove the quotation marks

Reply:
done